Documentation/concepts/networking/ipam/cluster-pool.rst

```diff
@@ -57,3 +57,76 @@ Check the ``Error`` field in the ``Status.Operator`` field:
 .. code-block:: shell-session
 
     kubectl get ciliumnodes -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.operator.error}{"\n"}{end}'
+
+**********************
+Cluster Pool v2 (Beta)
+**********************
+
+Cluster Pool v2 (Beta) extends the above mechanism to allow additional PodCIDRs
+to be dynamically allocated to each node based on usage. With v2, each Cilium
+agent instance reports the utilization of its PodCIDRs via the ``CiliumNode``
+resource.
+
+If a node is running low on available pod IPs, the operator will assign an
+additional PodCIDR to that node. Likewise, if a node has unused PodCIDRs, it
+will eventually release it, allowing the operator to re-assign the released
+PodCIDR to a different node if needed.
+
+When running v2, the ``CiliumNode`` resource is extended with an additional
+PodCIDR status section:
+
++-------------------------+----------------------------------------------------+
+|Field                    | Description                                        |
++=========================+====================================================+
+|``Spec.IPAM.PodCIDRs``   | List of assigned IPv4 and/or IPv6 PodCIDRs         |
++-------------------------+----------------------------------------------------+
+|``Status.IPAM.PodCIDRs`` | PodCIDR utilization                                |
+|                         | (one of: ``in-use``, ``depleted``, or ``released``)|
++-------------------------+----------------------------------------------------+
+
+The operator assigns a new PodCIDR to a node if all of its PodCIDRs are either
+depleted or released.
+
+Limitations
+===========
+
+Cluster Pool v2 is a preview feature. The following limitations currently apply
+to Cilium running in ``cluster-pool-v2beta`` IPAM mode:
+
+.. warning::
+  - Tunnel mode is not supported. Cluster Pool v2 may only be used in direct
+    routing mode.
+  - Transparent encryption with IPSec is not supported.
```

Comment:
Does this mean Wireguard is supported?

Reply:
Yes, WireGuard should work (I don't think I have tested it however). The reason why IPSec (and tunnel mode for that matter) are not supported at all are that there is a lot of code assuming that there is a single PodCIDR per node. Why v2, we can have multiple PodCIDRs.

That's probably fixable (IPSec has an escape hatch for AWS/Azure where there are no PodCIDRs at all), but I did not have the cycles to investigate yet how to adapt it.

Reply:
Given that you had this context, I wonder if it's worth opening an issue to track it, and linking to it in the warning

Reply:
Good idea. I'll add a link to #18987 for all potential future additions